.github/workflows/heartbeat-v4.yml

```diff
@@ -0,0 +1,138 @@
+name: heartbeat-v4
+on:
+  workflow_call: # As per documentation, the colon is necessary even though no config is required.
+  workflow_dispatch: # As per documentation, the colon is necessary even though no config is required.
+
+jobs:
+  production-heartbeat:
+    strategy:
+      # By default, if any job in a matrix fails, all other jobs are immediately cancelled. This makes the jobs run to completion instead.
+      fail-fast: false
+      matrix:
+        os: [{vm: ubuntu-latest, exe: .sh}, {vm: windows-2019, exe: .cmd}]
+        node: ['lts/*']
+    runs-on: ${{ matrix.os.vm }}
+    timeout-minutes: 60
+    steps:
+      # === Setup. We need to get the code, set up nodejs, and create the results directory. ===
+      - uses: actions/checkout@v4
+        with:
+          ref: 'main-4'
```

Comment:
In the original workflow, this was `release`, which was a mistake. It should have been `main`, since that was the production branch for v4. This hasn't caused any issues since there have been no breaking changes to v4 in many months, but should be corrected here.